### PR TITLE
[Android] UI tests for InputTransparent and IsEnabled fixes

### PR DESCRIPTION
**Pull request by adrianknight89**
### Description of Change

I'm able to run Android UITests now and wanted to submit tests for #344. In the process, I realized this needed a better fix.

### Bugs Fixed

- https://bugzilla.xamarin.com/show_bug.cgi?id=46458

### PR Checklist
- [x] Has tests (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard
- [x] Consolidate commits as makes sense

### CODE REVIEW

File: Xamarin.Forms.Platform.Android/VisualElementRenderer.cs [hunk 1]
```diff
@@ -87,14 +87,20 @@ void IOnClickListener.OnClick(AView v)
 
 		public override bool OnInterceptTouchEvent(MotionEvent ev)
 		{
-			if (Element.InputTransparent && Element.IsEnabled)
-				return false;
+			if (!Element.IsEnabled || Element.InputTransparent)
+				return true;
```
Comment (adrianknight89) [inline Xamarin.Forms.Platform.Android/VisualElementRenderer.cs]: Touches seem to be put on a stack from parent to inner most child and this stack is emptied out from that child back to parent. Makes sense. Here, the parent says consume the event and do not send it to children if I'm disabled or input transparent.

**Comment by jassmith** [comment]
This is a great start, we need to define in full what this behavior should be on each platform. Traditionally we have left it up to the underlying platform however it is probably time we took direct control of it.

**Comment by adrianknight89** [comment]
@jassmith Have you guys discussed this internally yet? Would be great if we could give this priority. Ideally, the behavior for transparency and enabled states should be the same across all platforms.

**Comment by rmarinho** [comment]
@hartez do you think can we import to our branch?

**Comment by hartez** [comment]
These tests (and the Android fixes) are now merged as part of [PR 808](https://github.com/xamarin/Xamarin.Forms/pull/808).

**Pull request closed without merging**